src/idl_gen_swift.cpp

```diff
@@ -0,0 +1,721 @@
+    code_ +=
+        "\tpublic static func start{{STRUCTNAME}}(_ fbb: FlatBufferBuilder) -> "
+        "UOffset { fbb.startTable(with: {{NUMBEROFFIELDS}}) }";
+
+    std::string builder_string = ", _ fbb: FlatBufferBuilder) { fbb.add(";
+    std::vector<std::string> require_fields;
+
+    for (auto it = struct_def.fields.vec.begin();
+         it != struct_def.fields.vec.end(); ++it) {
+      auto &field = **it;
+      if (field.deprecated) continue;
+      if (field.key) key_field = &field;
+
+      auto offset = NumToString(field.value.offset);
+      auto name = Name(field);
+      auto type = GenType(field.value.type);
+      if (field.required) require_fields.push_back(offset);
+      code_.SetValue("VALUENAME", name);
+      code_.SetValue("VALUETYPE", type);
+      code_.SetValue("OFFSET", NumToString(it - struct_def.fields.vec.begin()));
+      code_.SetValue("CONSTANT", field.value.constant);
+      code_ += "\tpublic static func add({{VALUENAME}}: \\";
+      if (IsScalar(field.value.type.base_type) &&
+          !IsBool(field.value.type.base_type)) {
+        std::string is_enum = IsEnum(field.value.type) ? ".rawValue" : "";
+        code_ += "{{VALUETYPE}}" + builder_string + "element: {{VALUENAME}}" +
+                 is_enum + ", def: {{CONSTANT}}, at: {{OFFSET}}) }";
+      } else if (IsBool(field.value.type.base_type)) {
+        code_.SetValue("VALUETYPE", "Bool");
+        code_.SetValue("CONSTANT",
+                       "0" == field.value.constant ? "false" : "true");
+        code_ +=
+            "{{VALUETYPE}}" + builder_string +
+            "condition: {{VALUENAME}}, def: {{CONSTANT}}, at: {{OFFSET}}) }";
+      } else {
+        std::string offset_type = field.value.type.base_type == BASE_TYPE_STRING
+                                      ? "Offset<String>"
+                                      : "Offset<UOffset>";
+        std::string reader_type =
+            IsStruct(field.value.type) && field.value.type.struct_def->fixed
+                ? "structOffset: {{OFFSET}}) }"
+                : "offset: {{VALUENAME}}, at: {{OFFSET}})  }";
+        code_ += offset_type + builder_string + reader_type;
+      }
+    }
+    code_ +=
+        "\tpublic static func end{{STRUCTNAME}}(_ fbb: FlatBufferBuilder, "
+        "start: "
+        "UOffset) -> Offset<UOffset> { let end = Offset<UOffset>(offset: "
+        "fbb.endTable(at: start))\\";
+    if (require_fields.capacity() != 0) {
+      std::string fields = "";
+      for (auto it = require_fields.begin(); it != require_fields.end(); ++it)
+        fields += *it + ", ";
+      code_.SetValue("FIELDS", fields.substr(0, fields.size() - 2));
+      code_ += "; fbb.require(table: end, fields: [{{FIELDS}}])\\";
+    }
+    code_ += "; return end }";
+
+    std::string spacing = "\t\t";
+    if (!struct_def.fixed && struct_def.has_key) {
+      code_.SetValue("VALUENAME", struct_def.name);
+      code_.SetValue("VOFFSET", NumToString(key_field->value.offset));
```

Comment:
is key_field not-null here? what will happen if all fields are deprecated?

Reply:
so you think a check would be nice before entering the if statement? 
`if (key_field != nullptr && !struct_def.fixed && struct_def.has_key) {`